browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -406,18 +416,21 @@ void SpeedreaderTabHelper::DOMContentLoaded(
         getElementById('c93e2206-2f31-4ddc-9828-2bb8e8ed940e');
       if (!link)
         return;
-      link.text = '$1';
+      link.text = "$1";
       link.addEventListener('click', (e) => {
         window.speedreader.showOriginalPage();
       })
     })();
   )js";
 
-  const auto link_text = brave_l10n::GetLocalizedResourceUTF16String(
+  auto link_text = brave_l10n::GetLocalizedResourceUTF16String(
       IDS_SPEEDREADER_SHOW_ORIGINAL_PAGE_LINK);
+  if (test::g_show_original_link_title) {
+    link_text = *test::g_show_original_link_title;
+  }
+
   // Make sure that the link text doesn't contain js injection
-  CHECK_EQ(std::u16string::npos, link_text.find(u'\''));
-  CHECK_EQ(std::u16string::npos, link_text.find(u'\\'));
+  base::ReplaceChars(link_text, u"\"", u"\\\"", &link_text);
```

Comment:
does it get escaped here if `link_text` is `\u0022hello\u0022` ?

Reply:
Yes. I also added this case into the test.